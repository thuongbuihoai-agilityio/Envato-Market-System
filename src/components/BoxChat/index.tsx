import { memo } from 'react';
import { Box, Image, Heading, Flex, Input, Button } from '@chakra-ui/react';

// Constants
import { IMAGES } from '@app/constants';

// Components
import Message from '@app/components/BoxChat/Message';

// Mocks
import { MESSAGE_TIME } from '@app/mocks';

const BoxChatComponent = (): JSX.Element => (
  <Box w="full" bg="background.body.quaternary" borderRadius="lg">
    <Flex
      direction="row"
      justifyContent="space-between"
      alignItems="center"
      justify="center"
      borderBottom="1px solid"
      borderBottomColor="border.tertiary"
      padding="24px 26px"
    >
      <Heading
        as="h3"
        fontWeight="semibold"
        color="text.primary"
        fontSize="2xl"
        textTransform="capitalize"
      >
        team chat
      </Heading>

      <Flex gap={5}>
        <Image
          src={IMAGES.USER_AVATAR.url}
          alt={IMAGES.USER_AVATAR.alt}
          w="52px"
          h={8}
        />

        <Button
          aria-label="btn-icon-plus"
          bg="secondary.800"
          w={9}
          h={9}
          borderRadius="50%"
        >
          <Image
            src="icons/plus.svg"
            alt={IMAGES.USER_AVATAR.alt}
            w={3.5}
            h={3.5}
          />
        </Button>
      </Flex>
    </Flex>

    <Box padding={{ base: '24px 20px', lg: '38px 35px' }}>
      <Message
        content="Hi, What can i help you with?"
        avatarPosition="before"
        avatar={IMAGES.CHAT_USER_AVATAR.url}
        localeTime={MESSAGE_TIME}
      />

      <Message
        avatar={IMAGES.CHAT_USER_AVATAR.url}
        isImage
        localeTime={MESSAGE_TIME}
      />

      <Flex direction="row-reverse">
        <Message
          content="Hello, I want to know more about your product"
          avatarPosition="after"
          isOwnerMessage
          avatar={IMAGES.CHAT_USER_AVATAR.url}
          localeTime={MESSAGE_TIME}
        />
      </Flex>

      <Message
        content="Sure, I can help you with that"
        avatarPosition="before"
        avatar={IMAGES.CHAT_USER_AVATAR.url}
        localeTime={MESSAGE_TIME}
      />

      <Flex justify="center" align="center">
        <Box
          border="1px solid"
          borderColor="border.secondary"
          p="0 20px"
          mt={5}
          borderRadius="lg"
          w="full"
        >
          <Flex direction="row" alignItems="center">
            <Image
              src={IMAGES.ATTACH.url}
              alt={IMAGES.ATTACH.alt}
              cursor="pointer"
            />

            <Input
              variant="authentication"
              ml={5}
              type="number"
              _dark={{
                border: 'none',
              }}
              sx={{ border: 'none', padding: 0 }}
              placeholder="Type your message here"
            />

            <Image
              src={IMAGES.MICRO.url}
              alt={IMAGES.MICRO.alt}
              cursor="pointer"
            />
          </Flex>
        </Box>
        <Image
          mt={5}
          ml={5}
          src={IMAGES.SEND.url}
          alt={IMAGES.SEND.alt}
          cursor="pointer"
        />
      </Flex>
    </Box>
  </Box>
);

const BoxChat = memo(BoxChatComponent);

export default BoxChat;
