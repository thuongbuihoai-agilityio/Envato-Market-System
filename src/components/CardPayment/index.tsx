import { memo } from 'react';
import { Box, Heading } from '@chakra-ui/react';
import { SubmitHandler } from 'react-hook-form';

// Hooks
import { useForm } from '@app/hooks';

// Components
import { CardBalance } from './CardBalance';
import { UserSelector } from './UserSelector';
import { EnterMoney } from './EnterMoney';

export interface CardPaymentProps {
  balance?: number;
  onSubmit?: SubmitHandler<TTransfer>;
}

export type TTransfer = {
  amount: string;
  userId: string;
};

const CardPaymentComponent = ({
  balance = 24098,
  onSubmit = () => {},
}: CardPaymentProps): JSX.Element => {
  const {
    control,
    handleSubmit,
    formState: { isValid, isSubmitting },
  } = useForm<TTransfer>({});

  return (
    <Box
      p={4}
      w="full"
      bg="background.body.quaternary"
      py={{ base: 4, md: 5 }}
      px={{ base: 4, md: 10 }}
      borderRadius="lg"
    >
      <Heading
        as="h3"
        fontWeight="bold"
        color="text.primary"
        fontSize="lg"
        mb={3}
        textTransform="capitalize"
      >
        my wallet
      </Heading>

      <CardBalance balance={balance} />

      <Box as="form" mt={4} onSubmit={handleSubmit(onSubmit)}>
        <UserSelector control={control} />
        <EnterMoney isDisabled={!isValid || isSubmitting} control={control} />
      </Box>
    </Box>
  );
};

const CardPayment = memo(CardPaymentComponent);

export default CardPayment;
