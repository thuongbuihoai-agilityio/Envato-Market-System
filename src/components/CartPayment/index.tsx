import { memo } from 'react';
import {
  Box,
  Text,
  Image,
  Select,
  Heading,
  Flex,
  Input,
  Button,
} from '@chakra-ui/react';

// Assets
import CARD_PAYMENT from 'src/assets/images/card-payment.svg';
import { ChevronIcon } from '@assets/images/ChevronIcon';

// Constants
import { IMAGES } from '@constants/index';

const CartPaymentComponent = (): JSX.Element => (
  <Box p={4} w="full" bg="background.body.quaternary" padding="20px 42px">
    <Heading
      as="h3"
      fontWeight="bold"
      color="text.primary"
      fontSize="lg"
      mb={3}
      textTransform="capitalize"
    >
      my wallet
    </Heading>

    <Image src={CARD_PAYMENT} alt="Payment Card" />

    <Box mt={4}>
      <Text
        fontWeight="bold"
        color="text.primary"
        fontSize="lg"
        mb={3}
        textTransform="capitalize"
      >
        quick transfer
      </Text>

      <Box position="relative">
        <Select
          size="lg"
          sx={{
            paddingLeft: '50px',
          }}
          borderColor="border.secondary"
          color="text.primary"
          icon={<ChevronIcon />}
        >
          <option value="debit" color="text.primary">
            Debit
          </option>
        </Select>

        <Image
          src={IMAGES.DEBIT_ICON.url}
          alt={IMAGES.DEBIT_ICON.alt}
          boxSize={6}
          position="absolute"
          left={5}
          top="50%"
          transform="translateY(-50%)"
        />

        <Text
          sx={{
            position: 'absolute',
            right: 10,
            top: '50%',
            transform: 'translateY(-50%)',
          }}
          fontWeight="bold"
          fontSize="sm"
        >
          $ 10,431
        </Text>
      </Box>

      <Box
        border="1px solid"
        borderColor="border.secondary"
        p={4}
        mt={5}
        borderRadius="lg"
      >
        <Text color="text.secondary"> Enter amount </Text>
        <Flex direction="row" alignItems="center">
          <Text color="text.primary" fontSize="2xl" fontWeight="bold">
            $
          </Text>
          <Input
            variant="authentication"
            type="number"
            _dark={{
              border: 'none',
            }}
            sx={{ border: 'none', padding: 0 }}
          />

          <Image
            src={IMAGES.USER_AVATAR.url}
            alt={IMAGES.USER_AVATAR.alt}
            boxSize={6}
            mt={3}
            w="42px"
          />
        </Flex>
      </Box>

      <Button mt={4} colorScheme="primary" fontWeight="bold">
        Send Money
      </Button>
    </Box>
  </Box>
);

const CardPayment = memo(CartPaymentComponent);

export default CardPayment;
