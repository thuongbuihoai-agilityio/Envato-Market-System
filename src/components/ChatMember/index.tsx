import { memo } from 'react';
import {
  Box,
  Flex,
  Text,
  Avatar,
  AvatarBadge,
  useColorModeValue,
} from '@chakra-ui/react';

// Themes
import { colors } from '@app/themes/bases/colors';

// Utils
import { getStatusColor } from '@app/utils';

export type Props = {
  avatar?: string;
  name?: string;
  status?: string;
  localeTime?: string;
  icon?: React.ReactNode;
  statusColor?: string;
  onClick?: () => void;
};

const ChatMember = ({
  avatar,
  name,
  localeTime,
  icon,
  statusColor = '',
  onClick,
}: Props) => {
  const colorFill = useColorModeValue(
    colors.secondary[200],
    colors.secondary[600],
  );

  return (
    <Box
      cursor="pointer"
      _hover={{ bg: colorFill }}
      onClick={onClick}
      borderRadius={'lg'}
    >
      <Flex justify="space-between" p={3.5}>
        <Flex gap={3}>
          <Avatar src={avatar} borderRadius="50%">
            <AvatarBadge boxSize={4} bg={getStatusColor(statusColor)} top={7} />
          </Avatar>
          <Box mr={6}>
            <Text>{name}</Text>
          </Box>
        </Flex>

        <Flex direction="column">
          <Text>{localeTime}</Text>
          {icon}
        </Flex>
      </Flex>
    </Box>
  );
};

const ChatMemberMemorized = memo(ChatMember);

export default ChatMemberMemorized;
