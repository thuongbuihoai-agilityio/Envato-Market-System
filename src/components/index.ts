export { default as InputField } from './common/InputField';
export { default as Logo } from './common/Logo';
export { default as Benefit } from './common/Benefit';
export { default as Button } from './common/Button';
export { default as Divider } from './common/Divider';
export { default as Menu } from './common/Menu';
export { default as IconButton } from './common/IconButton';
export { default as Avatar } from './common/Avatar';
export { default as Dropdown } from './common/Dropdown';
export { default as TotalCard } from './TotalCard';
export { default as Lazy } from './Lazy';
export { default as TotalListSkeleton } from './Skeleton/TotalListSkeleton';
export { default as Sidebar } from './Sidebar';
export { default as Navigation } from './common/Navigation';
export { default as Header } from './common/Header';
export { default as Select } from './common/Select';
export { default as Pagination } from './common/Pagination';
export { default as FilterUser } from './FilterUser';
export { default as Table, type TDataSource } from './common/Table';
export * from './common/Table/Body';
export { default as HeadCell } from './common/Table/HeadCell';
export { default as TotalList } from './TotalList';
export { default as CartPayment } from './CartPayment';
export { default as BoxChat } from './BoxChat';
export { default as Efficiency } from './Efficiency';
export { default as RevenueFlow } from './RevenueFlow';
export { default as OverallBalance } from './OverallBalance';
export { default as TotalBalance } from './TotalBalance';
export { default as ExpandSidebar } from './ExpandSidebar';
export { default as MiniSidebar } from './MiniSidebar';
