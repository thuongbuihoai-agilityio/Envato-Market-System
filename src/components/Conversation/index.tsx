import { memo, useMemo, useState } from 'react';
import { Box, Heading, Flex, theme, useColorModeValue } from '@chakra-ui/react';
import ReactQuill from 'react-quill';
import 'react-quill/dist/quill.snow.css';

// themes
import { colors } from '@app/themes/bases/colors';

// Constants
import { AVATAR_POSITION, IMAGES } from '@app/constants';

// Components
import { Message } from '@app/components';
import { ChatMemberMemorized } from '@app/components';
import { CustomButton } from '@app/components';
import { SendIcon } from '@app/components/Icons';

// Mocks
import { MESSAGE_TIME, MESSAGES } from '@app/mocks';

// Stores
import { authStore } from '@app/stores';

// Interfaces
import { MessageType } from '@app/interfaces/messages';

export type Props = {
  activeMember?: string;
  filteredMessages: MessageType[];
};

const Conversation = ({ filteredMessages }: Props) => {
  const avatarURL = authStore(
    (state): string | undefined => state.user?.avatarURL,
  );

  const username = authStore(
    ({ user }): string | undefined => `${user?.firstName} ${user?.lastName}`,
  );

  const [editorValue, setEditorValue] = useState<string>('');
  const colorFill = useColorModeValue(
    theme.colors.white,
    colors.secondary[400],
  );

  const messagesToShow = useMemo(
    () => (filteredMessages.length > 0 ? filteredMessages : MESSAGES),
    [filteredMessages],
  );

  return (
    <Box w="full" borderRadius="lg">
      <Flex
        direction="row"
        justifyContent="space-between"
        padding={{ base: '8px', lg: '24px 26px' }}
        bg="background.body.septenary"
      >
        <Heading
          as="h3"
          fontWeight="semibold"
          color="text.primary"
          fontSize="2xl"
          textTransform="capitalize"
        >
          <ChatMemberMemorized
            avatar={avatarURL}
            name={username}
            status="Online"
            statusColor="online"
          />
        </Heading>
      </Flex>

      <Box padding={{ base: '24px 20px', lg: '38px 35px' }}>
        {messagesToShow.map((message): JSX.Element => {
          const { isSend, isAudio, uid, content } = message;

          return (
            <Message
              key={uid}
              content={content}
              isImage={isAudio}
              isOwnerMessage={isSend}
              avatarPosition={
                isSend ? AVATAR_POSITION.AFTER : AVATAR_POSITION.BEFORE
              }
              avatar={IMAGES.CHAT_USER_AVATAR.url}
              localeTime={MESSAGE_TIME}
            />
          );
        })}

        <Flex direction="column" width="full">
          <ReactQuill
            value={editorValue}
            onChange={(value) => setEditorValue(value)}
            modules={{
              toolbar: [
                ['bold', 'italic', 'underline'],
                ['image', 'code-block'],
              ],
            }}
            style={{
              width: '100%',
              backgroundColor: colorFill,
            }}
            theme="snow"
          />

          <CustomButton
            w="unset"
            px={4}
            py={2.5}
            leftIcon={<SendIcon />}
            fontSize="md"
            fontWeight="semibold"
            bgColor="primary.600"
            mt={4}
            alignSelf="flex-end"
          >
            Send
          </CustomButton>
        </Flex>
      </Box>
    </Box>
  );
};

const ConversationMemorized = memo(Conversation);

export default ConversationMemorized;
