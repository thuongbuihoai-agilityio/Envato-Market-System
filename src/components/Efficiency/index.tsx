import { memo, useCallback } from 'react';
import isEqual from 'react-fast-compare';

// Components
import {
  Box,
  Flex,
  Heading,
  Text,
  theme,
  useColorModeValue,
} from '@chakra-ui/react';
import { Select } from '@app/components';
import { TOption } from '@app/components/common/Select';
import EfficiencyInfo from './EfficiencyInfo';
import EfficiencyRefetch from './Refetching';

// Icons
import { Arrow } from '@app/assets/icons';

// Constants
import { EFFICIENCY_OPTIONS } from '@app/constants';

// Types
import { IEfficiency } from '@app/interfaces';

interface EfficiencyProps extends IEfficiency {
  isLoading?: boolean;
  isExchangeRate?: boolean;
  isLoadingWhenSelect?: boolean;
  onChangeSelect: (data: TOption) => void;
}

const EfficiencyComponent = ({
  arrival,
  spending,
  statistical,
  isLoading = false,
  isLoadingWhenSelect = false,
  onChangeSelect,
}: EfficiencyProps) => {
  const colorFill = useColorModeValue(
    theme.colors.gray[400],
    theme.colors.white,
  );

  const renderTitle = useCallback(
    ({ label }: TOption) => (
      <Flex alignItems="center">
        <Text>{label}</Text>
        <Arrow color={colorFill} />
      </Flex>
    ),
    [colorFill],
  );

  return (
    <Box bg="background.component.primary" rounded="lg">
      <Flex
        py={4}
        px={5}
        borderBottom="1px"
        borderColor="border.primary"
        justifyContent="space-between"
      >
        <Heading variant="heading2Xl" as="h3">
          Efficiency
        </Heading>
        <Box w={102} h={21}>
          <Select
            options={EFFICIENCY_OPTIONS}
            size="sm"
            variant="no-background"
            renderTitle={renderTitle}
            onSelect={onChangeSelect}
          />
        </Box>
      </Flex>
      {isLoadingWhenSelect && isLoading ? (
        <EfficiencyRefetch />
      ) : (
        <EfficiencyInfo
          spending={spending}
          statistical={statistical}
          arrival={arrival}
        />
      )}
    </Box>
  );
};

const Efficiency = memo(EfficiencyComponent, isEqual);

export default Efficiency;
