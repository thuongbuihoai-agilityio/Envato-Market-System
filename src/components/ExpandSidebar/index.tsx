import { memo, useCallback } from 'react';
import isEqual from 'react-fast-compare';
import {
  Drawer,
  DrawerBody,
  DrawerContent,
  DrawerHeader,
  DrawerOverlay,
  Image,
  VStack,
  useMediaQuery,
} from '@chakra-ui/react';

// type
import { SidebarProps } from '@app/layouts/Sidebar';
import { TMenuItem } from '../common/Menu';

// components
import { Logo, Menu } from '@app/components';

// constants
import { SIDEBAR, IMAGES } from '@app/constants';

// mocks
import { EXPAND_SIDEBAR_MENU_LIST } from '@app/constants';

const ExpandSidebar = ({
  onClose,
  onOpen,
  isOpen,
  roleAdmin,
}: SidebarProps) => {
  const [isMobileAndTablet] = useMediaQuery('(max-width: 1732px)');

  const handleCloseSideBar = useCallback(() => {
    isMobileAndTablet && onOpen();
  }, [isMobileAndTablet, onOpen]);

  return (
    <Drawer
      placement="left"
      onClose={onClose}
      isOpen={isOpen}
      trapFocus={false}
      onOverlayClick={handleCloseSideBar}
      variant={{
        '4xl': 'clickThrough',
      }}
      blockScrollOnMount={false}
    >
      <DrawerOverlay data-testid="expand-overlay" />

      <DrawerContent
        maxW={SIDEBAR.EXPAND_SIDEBAR_WIDTH}
        display="block"
        bg="background.component.primary"
        maxH="full"
      >
        <DrawerHeader
          display="flex"
          alignItems="center"
          h={108}
          borderRightColor="border.secondary"
          borderRightWidth="1px"
          borderBottomColor="border.secondary"
          borderBottomWidth="1px"
          pl={12.5}
        >
          <Logo />

          {/* Close button of Expand sidebar */}
          <Image
            src={IMAGES.LEFT_ARROW.url}
            alt={IMAGES.LEFT_ARROW.alt}
            position="absolute"
            top="auto"
            right={0}
            onClick={onOpen}
            cursor="pointer"
            id="close-expand"
          />
        </DrawerHeader>

        <DrawerBody
          pt={3.5}
          pl={12}
          pb={50}
          pr={0}
          w="full"
          h="full"
          sx={{
            '&::-webkit-scrollbar': {
              display: 'none',
            },
            '&::-webkit-scrollbar-thumb': {
              display: 'none',
            },
          }}
        >
          <VStack pr={12.5} mb={9}>
            {EXPAND_SIDEBAR_MENU_LIST(roleAdmin as string).map((item) => (
              <Menu
                key={item.id}
                title={item.title}
                listItem={item.listItem as TMenuItem[]}
                onClickMenuItem={handleCloseSideBar}
              />
            ))}
          </VStack>
        </DrawerBody>
      </DrawerContent>
    </Drawer>
  );
};

const ExpandSidebarComponent = memo(ExpandSidebar, isEqual);
export default ExpandSidebarComponent;
