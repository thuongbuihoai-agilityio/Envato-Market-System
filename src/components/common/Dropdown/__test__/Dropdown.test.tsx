import { render } from '@testing-library/react';
import '@testing-library/jest-dom';
import { MemoryRouter } from 'react-router-dom';

// component
import { Dropdown } from '@app/components';

describe('Dropdown render', () => {
  const renderComponent = ({
    name,
    permission,
  }: {
    name?: string;
    permission?: string;
  }) =>
    render(
      <Dropdown
        name={name || 'John Doe'}
        permission={permission || 'Super Admin'}
      />,
      { wrapper: MemoryRouter },
    );

  it('Should render match with snapshot.', () => {
    const { container } = renderComponent({});
    expect(container).toMatchSnapshot();
  });

  it('Get Dropdown component', () => {
    const { getByTestId } = renderComponent({});

    const avatar = getByTestId('TestDropdown');
    expect(avatar).toBeTruthy();
  });

  it('Re-render', () => {
    const { getByTestId, rerender } = renderComponent({});

    const avatar = getByTestId('TestDropdown');
    expect(avatar).toBeTruthy();

    rerender(<Dropdown name="ABC" permission="BCD" />);
  });
});
