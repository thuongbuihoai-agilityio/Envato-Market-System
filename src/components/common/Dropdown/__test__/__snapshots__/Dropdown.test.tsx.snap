// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Dropdown render Should render match with snapshot. 1`] = `
<div>
  <div
    class="css-cwsnob"
  >
    <button
      aria-controls="menu-list-:r1:"
      aria-expanded="false"
      aria-haspopup="menu"
      class="chakra-button chakra-menu__menu-button css-14rg285"
      id="menu-button-:r1:"
      type="button"
    >
      <span
        class="css-xl71ch"
      >
        <div
          class="css-70qvj9"
        >
          <span
            class="chakra-avatar css-1gt5voo"
          >
            <svg
              aria-label=" avatar"
              class="chakra-avatar__svg css-16ite8i"
              role="img"
              viewBox="0 0 128 128"
            >
              <path
                d="M103,102.1388 C93.094,111.92 79.3504,118 64.1638,118 C48.8056,118 34.9294,111.768 25,101.7892 L25,95.2 C25,86.8096 31.981,80 40.6,80 L87.4,80 C96.019,80 103,86.8096 103,95.2 L103,102.1388 Z"
                fill="currentColor"
              />
              <path
                d="M63.9961647,24 C51.2938136,24 41,34.2938136 41,46.9961647 C41,59.7061864 51.2938136,70 63.9961647,70 C76.6985159,70 87,59.7061864 87,46.9961647 C87,34.2938136 76.6985159,24 63.9961647,24"
                fill="currentColor"
              />
            </svg>
          </span>
          <div
            class="css-21uplc"
          >
            <div
              class="css-1f25a08"
            >
              <div
                class="css-70qvj9"
              >
                <p
                  class="chakra-text css-8gj9cf"
                >
                  John Doe
                </p>
                <svg
                  fill="none"
                  height="24"
                  stroke="#1A202C"
                  viewBox="0 0 24 24"
                  width="24"
                  xmlns="http://www.w3.org/2000/svg"
                >
                  <path
                    d="M7 10L12 14L17 10"
                    stroke-linecap="round"
                    stroke-linejoin="round"
                    stroke-width="2"
                  />
                </svg>
              </div>
              <p
                class="chakra-text css-pfmkmr"
              >
                Super Admin
              </p>
            </div>
          </div>
        </div>
      </span>
    </button>
    <div
      class="css-0"
      style="visibility: hidden; position: absolute; min-width: max-content; inset: 0 auto auto 0;"
    >
      <div
        aria-orientation="vertical"
        class="chakra-menu__menu-list css-p2gi7y"
        data-testid="TestDropdown"
        id="menu-list-:r1:"
        role="menu"
        style="transform-origin: var(--popper-transform-origin); opacity: 0; visibility: hidden; transform: scale(0.8) translateZ(0);"
        tabindex="-1"
      >
        <button
          class="chakra-menu__menuitem css-bnm4c7"
          data-index="0"
          id="menu-list-:r1:-menuitem-:r2:"
          role="menuitem"
          tabindex="-1"
          type="button"
        >
          <div
            class="css-k008qs"
          >
            <svg
              fill="none"
              height="24"
              stroke="#1A202C"
              viewBox="0 0 24 24"
              width="24"
              xmlns="http://www.w3.org/2000/svg"
            >
              <path
                d="M12.1197 12.7805C12.0497 12.7705 11.9597 12.7705 11.8797 12.7805C10.1197 12.7205 8.71973 11.2805 8.71973 9.51047C8.71973 7.70047 10.1797 6.23047 11.9997 6.23047C13.8097 6.23047 15.2797 7.70047 15.2797 9.51047C15.2697 11.2805 13.8797 12.7205 12.1197 12.7805Z"
                stroke-linecap="round"
                stroke-linejoin="round"
                stroke-width="1.5"
              />
              <path
                d="M18.7398 19.3796C16.9598 21.0096 14.5998 21.9996 11.9998 21.9996C9.39977 21.9996 7.03977 21.0096 5.25977 19.3796C5.35977 18.4396 5.95977 17.5196 7.02977 16.7996C9.76977 14.9796 14.2498 14.9796 16.9698 16.7996C18.0398 17.5196 18.6398 18.4396 18.7398 19.3796Z"
                stroke-linecap="round"
                stroke-linejoin="round"
                stroke-width="1.5"
              />
              <path
                d="M12 22C17.5228 22 22 17.5228 22 12C22 6.47715 17.5228 2 12 2C6.47715 2 2 6.47715 2 12C2 17.5228 6.47715 22 12 22Z"
                stroke-linecap="round"
                stroke-linejoin="round"
                stroke-width="1.5"
              />
            </svg>
            <a
              class="chakra-text css-1t7utft"
              href="/"
            >
              My profile
            </a>
          </div>
        </button>
        <button
          class="chakra-menu__menuitem css-bnm4c7"
          data-index="1"
          id="menu-list-:r1:-menuitem-:r3:"
          role="menuitem"
          tabindex="-1"
          type="button"
        >
          <div
            class="css-k008qs"
          >
            <svg
              fill="none"
              height="24"
              stroke="#1A202C"
              viewBox="0 0 24 24"
              width="24"
              xmlns="http://www.w3.org/2000/svg"
            >
              <path
                d="M2 12V7C2 4.79086 3.79086 3 6 3H18C20.2091 3 22 4.79086 22 7V17C22 19.2091 20.2091 21 18 21H8M6 8L9.7812 10.5208C11.1248 11.4165 12.8752 11.4165 14.2188 10.5208L18 8M2 15H8M2 18H8"
                stroke-linecap="round"
                stroke-width="1.5"
              />
            </svg>
            <a
              class="chakra-text css-1t7utft"
              href="/"
            >
              Inbox
            </a>
          </div>
        </button>
        <button
          class="chakra-menu__menuitem css-bnm4c7"
          data-index="2"
          id="menu-list-:r1:-menuitem-:r4:"
          role="menuitem"
          tabindex="-1"
          type="button"
        >
          <div
            class="css-k008qs"
          >
            <svg
              fill="none"
              height="24"
              viewBox="0 0 24 24"
              width="24"
              xmlns="http://www.w3.org/2000/svg"
            >
              <path
                d="M15 10L13.7071 11.2929C13.3166 11.6834 13.3166 12.3166 13.7071 12.7071L15 14M14 12L22 12M6 20C3.79086 20 2 18.2091 2 16V8C2 5.79086 3.79086 4 6 4M6 20C8.20914 20 10 18.2091 10 16V8C10 5.79086 8.20914 4 6 4M6 20H14C16.2091 20 18 18.2091 18 16M6 4H14C16.2091 4 18 5.79086 18 8"
                stroke="#1A202C"
                stroke-linecap="round"
                stroke-width="1.5"
              />
            </svg>
            <a
              class="chakra-text css-1t7utft"
              href="/"
            >
              Logout
            </a>
          </div>
        </button>
        <hr
          aria-orientation="horizontal"
          class="chakra-divider css-19ob8qo"
        />
        <button
          class="chakra-menu__menuitem css-rm7jzp"
          data-index="3"
          id="menu-list-:r1:-menuitem-:r5:"
          role="menuitem"
          tabindex="-1"
          type="button"
        >
          <a
            class="chakra-text css-0"
            href="/"
          >
            Setting
          </a>
        </button>
        <button
          class="chakra-menu__menuitem css-rm7jzp"
          data-index="4"
          id="menu-list-:r1:-menuitem-:r6:"
          role="menuitem"
          tabindex="-1"
          type="button"
        >
          <a
            class="chakra-text css-0"
            href="/"
          >
            User
          </a>
        </button>
      </div>
    </div>
  </div>
</div>
`;
